src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs

```diff
@@ -231,5 +269,16 @@ private void CheckDisposed()
                 throw new ObjectDisposedException(this.GetType().ToString());
             }
         }
+
+        /// <summary>
+        /// This class will only be allocated if Telemetry is enabled.
+        /// We can't use HttpRequestMessage's own finalizer because it is not sealed.
+        /// The finalizer will run iff OnStopped/OnAborted/Dispose were never called.
+        /// This way we ensure that RequestStop is always called if we call RequestStart.
+        /// </summary>
+        private sealed class HttpRequestMessageFinalizer
+        {
+            ~HttpRequestMessageFinalizer() => HttpTelemetry.Log.RequestStop();
+        }
```

Comment:
Shouldn't this run as part of the response/content cleanup, not the request?

We have some other finalizers already (HttpConnection etc.) to catch when a response isn't disposed -- to avoid the overhead of finalizer here, can we merge this into that? Or, is the goal to have this work for any handler?